ops/pebble.py

```diff
@@ -386,11 +386,29 @@ def __repr__(self):
                 ).format(self=self)
 
 
+class Plan:
+    """Represents a Pebble plan (service configuration).
+
+    Currently this only holds the combined services, not the individual
+    layers. The "raw_yaml" attribute is set to a YAML string for access to the
+    exact YAML that Pebble's plan API sent over the wire.
+    """
+
+    def __init__(self, raw: str = None):
+        if raw is not None:
+            d = yaml.safe_load(raw)
+        else:
+            d = {}
+        self.raw_yaml = raw
```

Comment:
Do we have an existing pattern for naming such a textual representatino, either in the operator framework or for Python in general?  In Go it's likely Bytes.

In either case, I think we should move this to be a method, for two reasons: first, because we may end up transferring it as a structured payload eventually, and if we do that having the attribute here would mean we either need to pre-compute or at least transform it into a computed attribute. Then, because this is actually a duplicated version of what exists in services, which means if we ever make changes here we'll have data inconsistent.

So I suggest moving it into _raw, and having a method to request the rendering as yaml instead.

Reply:
We don't seem to have an existing pattern for attributes. A local variable `raw` is used a lot in `ops/charm.py`, but none of those classes store it as an attribute. It's a bit different here because I want the rich object, but also want access to the YAML that came over the wire from Pebble's `/v1/plan` endpoint.

What I've done is basically what you've suggested. I've called it `to_yaml()` for consistency with the other similar `to_dict()` and `to_yaml()` functions in `ops/pebble.py` (and from a quick search, these function names are relatively common in Python code in the wild).

I've also made `Plan.services` a read-only property for now, to avoid issues if people update the attribute and then expect `to_yaml` to update. This is a simplification -- later we can add more functionality, make it read/write, etc.